code/modules/antagonists/traitor/objectives/demoralise_assault.dm

```diff
@@ -0,0 +1,119 @@
+	if(attacker != handler.owner.current)
+		return
+
+	// We want some sort of damaging attack to trigger this, rather than shoves and non-lethals.
+	if(!(attack_flags & ATTACKER_DAMAGING_ATTACK))
+		return
+
+	attacks_inflicted++
+
+	if(attacks_inflicted == attacks_required)
+		succeed_objective()
+
+/datum/traitor_objective/target_player/assault/ungenerate_objective()
+	UnregisterSignal(target, COMSIG_ATOM_WAS_ATTACKED)
+	UnregisterSignal(target, COMSIG_LIVING_DEATH)
+	UnregisterSignal(target, COMSIG_QDELETING)
+
+	target = null
+
+/datum/traitor_objective/target_player/assault/generate_objective(datum/mind/generating_for, list/possible_duplicates)
+	var/list/already_targeting = list() //List of minds we're already targeting. The possible_duplicates is a list of objectives, so let's not mix things
+	for(var/datum/objective/task as anything in handler.primary_objectives)
+		if(!istype(task.target, /datum/mind))
+			continue
+		already_targeting += task.target //Removing primary objective kill targets from the list
+
+	var/list/possible_targets = list()
+
+	for(var/datum/mind/possible_target as anything in get_crewmember_minds())
+		if(possible_target in already_targeting)
+			continue
+
+		if(possible_target == generating_for)
+			continue
+
+		if(!ishuman(possible_target.current))
+			continue
+
+		if(possible_target.current.stat == DEAD)
+			continue
+
+		if(possible_target.has_antag_datum(/datum/antagonist/traitor))
+			continue
+
+		possible_targets += possible_target
+
+	for(var/datum/traitor_objective/target_player/objective as anything in possible_duplicates)
+		possible_targets -= objective.target?.mind
+
+	if(generating_for.late_joiner)
+		var/list/all_possible_targets = possible_targets.Copy()
+		for(var/datum/mind/possible_target as anything in all_possible_targets)
+			if(!possible_target.late_joiner)
+				possible_targets -= possible_target
+		if(!possible_targets.len)
+			possible_targets = all_possible_targets
+
+	if(!possible_targets.len)
+		return FALSE
+
+	var/datum/mind/target_mind = pick(possible_targets)
+
+	target = target_mind.current
+	replace_in_name("%TARGET%", target.real_name)
+	replace_in_name("%JOB TITLE%", target_mind.assigned_role.title)
+
+	attacks_required = rand(min_attacks_required, max_attacks_required)
+	replace_in_name("%COUNT%", attacks_required)
+
+	RegisterSignal(target, COMSIG_LIVING_DEATH, PROC_REF(on_target_death))
+	RegisterSignal(target, COMSIG_QDELETING, PROC_REF(on_target_qdeleted))
+
+	return TRUE
+
+/datum/traitor_objective/target_player/assault/proc/on_target_qdeleted()
+	SIGNAL_HANDLER
+	if(objective_state == OBJECTIVE_STATE_INACTIVE)
+		//don't take an objective target of someone who is already obliterated
+		fail_objective()
+
+/datum/traitor_objective/target_player/assault/proc/on_target_death()
+	SIGNAL_HANDLER
+	if(objective_state == OBJECTIVE_STATE_INACTIVE)
+		//don't take an objective target of someone who is already dead
+		fail_objective()
```

Comment:
I think you may want to do something here if the target dies/qdeletes anyways even if the objective is active.
Best thing to do is to probably just invalidate the objective regardless of whether it's inactive or not.